clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -6052,6 +6054,10 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   if (!RawTriple.isOSDarwin() && !RawTriple.isNVPTX())
     CmdArgs.push_back("-mconstructor-aliases");
 
+  // Assume -fno-strict-bool in the Darwin kernel.
+  if (KernelOrKext)
```

Comment:
For this you should also add a Driver test that confirms that the flag is passed to the fronted when the driver is passed -mkernel or -fapple-kext.